### PR TITLE
feat: :chart_with_upwards_trend: integrate sentry

### PR DESCRIPTION
**Pull request by RyanHirsch**
Add sentry tracking for client side errors, this can be considered the trial run of the product.
The token required during build can be seen here - https://sentry.io/settings/podverse/developer-settings/git-hub-37f39d/

### CODE REVIEW

**Comment by RyanHirsch** [comment]
🎉

**Comment by RyanHirsch** [comment]
@mitchdowney I think I have the build working with sentry (and that is, don't upload source maps in development which was the root cause of the issue). However, it should upload source maps in the staging action but I haven't tested that.

It looks like the master action just pulls the code from the container registry so it shouldn't be impacted in any way.

It'd probably be best for someone else to run this on their dev box to make sure it didn't stash any files I'm unaware of with credentials on my box when I ran the config script.

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -21,6 +21,12 @@ jobs:
         uses: actions/setup-node@v2
 
       - name: Build
+        env:
+          SENTRY_ORG: podverse
+          SENTRY_PROJECT: react
+          SENTRY_AUTH_TOKEN: 4b0697089ef64bf9922dd4697797627f366751ab589b43efb037a41b916cff36
```
Comment (mitchdowney) [inline .github/workflows/develop.yml:27]: Should this be a secret?

File: pages/_error.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  // indicate a bug introduced in Next.js, so record it in Sentry
+  Sentry.captureException(
+    new Error(`_error.js getInitialProps missing data at path: ${asPath}`),
+  );
+  await Sentry.flush(2000);
+
+  return errorInitialProps;
+};
+
+export default MyError;
```
Comment (mitchdowney) [inline pages/_error.js:65]: Very nice.

File: sentry.client.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// This file configures the initialization of Sentry on the browser.
+// The config you add here will be used whenever a page is visited.
+// https://docs.sentry.io/platforms/javascript/guides/nextjs/
+
+import * as Sentry from '@sentry/nextjs';
+
+const SENTRY_DSN = process.env.SENTRY_DSN || process.env.NEXT_PUBLIC_SENTRY_DSN;
+
+Sentry.init({
+  dsn: SENTRY_DSN || 'https://ee0783ce357641a3bd5e692f5f3aaa21@o1107338.ingest.sentry.io/6134404',
```
Comment (mitchdowney) [inline sentry.client.config.js:10]: Not familiar with this...this URL doesn't have to be a secret?

Comment (RyanHirsch) [reply sentry.client.config.js:10]: nope, its client side, the client can't be trusted, blah, blah, blah.

Technically, if someone wanted to be malicious they should start sending us fake errors, but thats true with any client side web tracking.

File: .github/workflows/develop.yml (hunk 1 above)
Comment (RyanHirsch) [reply .github/workflows/develop.yml:27]: it can't do anything but read some project settings, and shouldn't even be needed anymore, I'll kill it.

**Review by mitchdowney — APPROVED** [review]

**Pull request merged**